File: bindings/imgui_bundle/demos_python/demos_tex_inspect/demo_tex_inspect_simple.py
# demo_tex_inspect_simple
# See equivalent C++ program: demos_cpp/demos_tex_inspect/demo_tex_inspect_simple.cpp
from imgui_bundle import imgui_tex_inspect, ImVec2, immapp, hello_imgui

from imgui_bundle.demos_python.demo_utils.api_demos import (
    set_hello_imgui_demo_assets_folder,
)


@immapp.static(texture_id=None)
def demo_gui():
    static = demo_gui

    if static.texture_id is None:
        static.texture_id = hello_imgui.im_texture_id_from_asset(
            "images/bear_transparent.png"
        )

    texture_size = ImVec2(512.0, 512.0)

    flags = 0
    inspector_size = imgui_tex_inspect.SizeIncludingBorder(immapp.em_to_vec2(30, 30))
    if imgui_tex_inspect.begin_inspector_panel(
        "Texture Inspector", static.texture_id, texture_size, flags, inspector_size
    ):
        pass # nothing to do here
    imgui_tex_inspect.end_inspector_panel() # Always call end_inspector_panel() even if begin_inspector_panel() returns false


def main():
    set_hello_imgui_demo_assets_folder()
    immapp.run(
        demo_gui, with_tex_inspect=True, with_markdown=True, window_size=(1200, 1000)
    )


if __name__ == "__main__":
    main()

File: bindings/imgui_bundle/demos_cpp/demos_tex_inspect/demo_tex_inspect_simple.cpp
// Demo imgui_tex_inspect
// See equivalent python program: bindings/imgui_bundle/demos/demos_tex_inspect/demo_tex_inspect.py
#include "immapp/immapp.h"
#ifdef IMGUI_BUNDLE_WITH_TEXT_INSPECT
#include "imgui_tex_inspect/imgui_tex_inspect.h"


void demo_tex_inspect_simple()
{
    static ImTextureID textureId = 0;
    static ImVec2 textureSize(512.f, 512.f);
    if (textureId == 0)
        textureId = HelloImGui::ImTextureIdFromAsset("images/bear_transparent.png");

    ImGui::Text("Simple Demo");
    ImGuiTexInspect::InspectorFlags  flags = 0;
    ImGuiTexInspect::SizeIncludingBorder inspectorSize(ImmApp::EmToVec2(40.f, 40.f));

    if (ImGuiTexInspect::BeginInspectorPanel(
        "Texture inspector",
        textureId,
        textureSize, flags,
        inspectorSize))
    {
        // nothing to do here
    }
    ImGuiTexInspect::EndInspectorPanel(); // Should be called even if BeginInspectorPanel returns false
}


#include "demo_utils/api_demos.h"
// A main() is provided automatically bu demos_cpp/_auto_main/
// However, the version below is left as a documentation reference,
// since imgui_tex_inspect requires to be initialized at startup
int disabled_main()
{
    HelloImGui::SetAssetsFolder(DemosAssetsFolder());

    ImmApp::Run(
        HelloImGui::SimpleRunnerParams {
            .guiFunction=demo_tex_inspect_simple,
            .windowTitle = "demo_tex_inspect_simple",
            .windowSize = {1000, 800}
        },
        ImmApp::AddOnsParams {
            .withTexInspect = true
        }
    );
    return 0;
}

#else // #ifdef IMGUI_BUNDLE_WITH_TEXT_INSPECT
void demo_tex_inspect_simple()
{
    ImGui::Text("This demo requires IMGUI_BUNDLE_WITH_TEXT_INSPECT to be enabled");
}
#endif